src/NServiceBus.Core.Tests/Routing/SingleInstanceRoundRobinDistributionStrategyTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿namespace NServiceBus.Core.Tests.Routing
+{
+    using System.Collections.Generic;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class SingleInstanceRoundRobinDistributionStrategyTests
```

Comment:
Should there be a test here that makes sure the round robin is wrapping back to the first instance correctly? So for n instances, the n + 1 call to SelectDestination() should return the first instance again?